operator/v1/types_ingress.go

```diff
@@ -0,0 +1,219 @@
+//
+// When an IngressController is created, a new ingress controller deployment is
+// created to allow external traffic to reach the services that expose Ingress
+// or Route resources. Updating this resource may lead to disruption for public
+// facing network connections as a new ingress controller revision may be rolled
+// out.
+//
+// https://kubernetes.io/docs/concepts/services-networking/ingress-controllers
+//
+// Additional features are available by default and through explicit
+// configuration.
+//
+// 1. To publish the ingress controller endpoints to the Internet, other
+//    networks, or external load balancers, configure an endpoint publishing
+//    strategy.
+//
+// 2. On certain cloud platforms, when publishing an ingress controller, managed
+//    wildcard DNS is automatically enabled. OpenShift will manage a DNS record
+//    pointing to the ingress controller. DNS records are managed only in DNS
+//    zones defined in the DNS cluster configuration resource.
+//
+// 3. If an ingress controller does not specify a default certificate, a new
+//    self-signed certificate valid for the specified domain is generated for
+//    the ingress controller.
+//
+// Whenever possible, sensible defaults for the platform are used. See each
+// field for more details.
+type IngressController struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the IngressController.
+	Spec IngressControllerSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the IngressController.
+	Status IngressControllerStatus `json:"status,omitempty"`
+}
+
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type IngressControllerSpec struct {
+	// domain is a DNS name used to configure various features that help publish
+	// the ingress controller and enable external integrations.
+	//
+	// * The value is published to individual Route statuses so that end-users
+	//   know where to target external DNS records.
+	//
+	// * When wildcard DNS management is enabled, domain is the base domain used
+	//   to construct the wildcard host name.
+	//
+	// * If a generated default certificate is used for the ingress controller,
+	//   the certificate will be valid for domain.
+	//
+	// domain must be unique among all IngressControllers, and cannot be
+	// updated.
+	//
+	// If empty, defaults to the cluster Ingress config domain.
+	//
+	// +optional
+	Domain string `json:"domain,omitempty"`
+
+	// replicas is the desired number of ingress controller replicas. If unset,
+	// defaults to 2.
+	//
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// endpointPublishingStrategy is used to publish the ingress controller
+	// endpoints to other netowrks, enable load balancer integrations, etc.
+	//
+	// If empty, the default is based on the cluster platform:
```

Comment:
Enumerate all possible strategies in this comment (the godoc on the constant doesn't show up in oc explain)

Reply:
Actually, don't, do it on `type`.

Reply:
Done